cmd/forwarder/fixer.go

```diff
@@ -42,10 +42,15 @@ func fix(r io.Reader, remoteAddr string, logplexDrainToken string) ([]byte, erro
 		messageWriter.Write(header.Procid)
 		messageWriter.WriteString(" ")
 		messageWriter.Write(header.Msgid)
+
+		if authUser != "" {
```

Comment:
This doesn't work here. You aren't in the message yet and should be starting the `STRUCTURED-DATA` section of the syslog frame, which is started down on line 51 (was 45) by writing ` [origin ip="` and finishes on line 53 (was 47) when `"]` is written.

This needs to move under line 62, but before the first `Write*` to `messageLenWriter`.

Or did you intend to stick this in the `STRUCTURED-DATA` section? I'm not sure if splunk surfaces that info in any way? If it does that may be better, but requires other formatting changes as structured data is [well defined](https://tools.ietf.org/html/rfc5424#section-6.3).

Reply:
I originally put it at the end -- see [this commit](https://github.com/heroku/log-iss/pull/23/commits/99c66295b87830c290f99d8941dd404a43d783fa#diff-1209027da991fd7acd1b7c3a08f61706R59) -- however, in the tests, some lines have new line characters (`\n`) at the end of the line resulting in something like `... foo=bar\n log_iss_user=ingest`.

I moved it forward in the line, (admittedly) knowing nothing about syslog formatting or it's structured data. I read up a bit on it (thanks for that link) and given that this is only going to be turned on for a bit and maybe only on a single indexer, I think it's overkill. The only solution I can come up with then is to check the last character of the line line and if it's an new line character, remove it. I'm worried though, about the overhead of doing this on every line and ensuring that I don't missing something. I think it's safe to assume it will always be `\n` and am moving forward with that assumption, but I'm not 100% sure.

Updated commit in coming.